File: Java/BOJ/21921 블로그.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        int N = Integer.parseInt(st.nextToken());
        int X = Integer.parseInt(st.nextToken());
        int[] blog = new int[N];
        st = new StringTokenizer(br.readLine());
        int maxValue = Integer.MIN_VALUE;
        for(int i = 0; i < N; i++) {
        	int num = Integer.parseInt(st.nextToken());
        	blog[i] = num;
        	if (num > maxValue) {
        		maxValue = num;
        	}
        }
        if (maxValue == 0) {
            System.out.println("SAD");
        } else {
            int x = 0;
            int value = 0;
            int result = 1;

            for (int i = 0; i < X; i++) {
                x += blog[i];
            }
            value = x;

            for (int i = X; i < N; i++) {
                value -= blog[i - X];
                value += blog[i];
                if (value > x) {
                    x = value;
                    result = 1;
                } else if (value == x) {
                    result++;
                }
            }

            System.out.println(x);
            System.out.println(result);
        }
    }
}

File: Python/BOJ/21921 블로그.py
import sys
N, X = map(int, sys.stdin.readline().split())
blog = list(map(int, sys.stdin.readline().split()))
if max(blog) == 0:
    print('SAD')
else:
    x = sum(blog[:X])
    value = x
    result = 1
    for i in range(X, N):
        value -= blog[i-X]
        value += blog[i]
        if value > x:
            x = value
            result = 1
        elif value == x:
            result += 1
    print(x)
    print(result)
